kernel/inductive.ml

```diff
@@ -1113,6 +1113,73 @@ let filter_stack_domain env nr p stack =
   in
   filter_stack env ar stack
 
+let merge_filter l l' =
+  let n = List.length l in
+  let n' = List.length l' in
+  if n < n' then List.map2 (&&) l (List.firstn n l')
+  else List.map2 (&&) (List.firstn n' l) l'
```

Comment:
It's a weird way to write a function that can be written much more elegantly with a pattern-matching.

Reply:
Indeed!